vlasiator.cpp

```diff
@@ -408,6 +408,8 @@ int main(int argn,char* args[]) {
    FsGrid< std::array<Real, fsgrids::ehall::N_EHALL>, 2> EHallGrid(fsGridDimensions, comm, periodicity,gridCoupling);
    FsGrid< std::array<Real, fsgrids::egradpe::N_EGRADPE>, 2> EGradPeGrid(fsGridDimensions, comm, periodicity,gridCoupling);
    FsGrid< std::array<Real, fsgrids::moments::N_MOMENTS>, 2> momentsGrid(fsGridDimensions, comm, periodicity,gridCoupling);
+   FsGrid< std::array<Real, fsgrids::moments::N_MOMENTS>, 2> swapGrid(fsGridDimensions, comm, periodicity,gridCoupling);
```

Comment:
I don't think that we need to drag it along from here and pass it down, it can be declared with the = operator on the spot. Saves some memory.